File: Test/test_cases/python3/tests/Test4.py
from Buyer import Buyer
from Seller import Seller

class Test:
	def test(self):
		buyer = Buyer(2)
		seller = Seller(10)
		score = 0

		buyer.buy(seller)
		if buyer.getCash()==1 and seller.getQuantity() == 9:
			seller.setQuantity(0)
			buyer.buy(seller)
			if buyer.getCash()==1:
				score = 1

		return score

File: Test/test_cases/java/tests/Test4.java
import java.io.*;
class Test
{
  public int test() throws Exception
  {
    Buyer x1=new Buyer(2);
    Seller x=new Seller(10);
    int score=0;
    x1.buy(x);
    if(x1.getCash()==1)
    {
      x1.setCash(0);
      x1.buy(x);
      if(x1.getCash()==0)
      {
        x1.setCash(1);
        x.setQuantity(0);
        x1.buy(x);
        if(x1.getCash()==1)
        {
          score=1;
        }
      }
    }
    return score;
  }
}
